app/services/bulk_upload/lettings/year2023/row_parser.rb

```diff
@@ -402,7 +402,7 @@ class BulkUpload::Lettings::Year2023::RowParser
 
   validate :validate_nulls, on: :after_log
 
-  validate :validate_uprn_exists_if_any_key_address_fields_are_blank, on: :after_log
+  validate :validate_uprn_exists_if_any_key_address_fields_are_blank, on: :after_log, unless: -> { supported_housing? }
```

Comment:
tiny thing, making this condition `if {: -> { general_needs? }`  instead seems a bit clearer to me but I accept that it doesn't make any functional difference because we couldn't get past the setup section if needstype was blank

Reply:
Good point, I think we're checking the value of field 4 instead of the bulk upload needstype, so if that's not given we want to run this as well. That was my thought process for doing an unless, but I don't think it matters much cause of a similar reason you mentioned, given a setup field would be missing